app/assets/javascripts/discourse/lib/plugin-api.js.es6

```diff
@@ -867,6 +870,19 @@ class PluginApi {
     addComposerUploadHandler(extensions, method);
   }
 
+  /**
+   * Registers a function to generate Markdown after a file has been uploaded.
+   *
+   * Example:
+   *
+   * api.addComposerUploadMarkdownGetter(upload => {
+   *   return `_uploaded ${upload.original_filename}_`;
+   * })
+   */
+  addComposerUploadMarkdownGetter(getter) {
```

Comment:
Maybe `addComposerUploadMarkdownResolver` ?

Also behavior when chained is a bit tricky, do we even want to allow chaining?

Reply:
It will use the last "truthy" value returned by a resolver. This way it allows multiple plugins to work correctly as long as an upload is not of interest to multiple plugins (or they could still alter attributes of the `upload` object such as filename).